docs/developer/index.rst

```diff
@@ -935,3 +935,187 @@ We can view the model plot object::
     ylabel = y
     title  = Model
     histo_prefs = {'yerrorbars': False, 'ecolor': None, ... , 'linecolor': None}
+
+
+.. _dataimg_coords:
+
+Coordinate conversion for image data
+------------------------------------
+
+The :py:class:`sherpa.data.Data2D` class provides basic support for
+fitting models to two-dimensional data; that is, data with two
+independent axes (called "x0" and "x1" although they should be
+accessed via the :py:attr:`~sherpa.data.Data2D.indep` attribute).  The
+:py:class:`sherpa.astro.data.DataIMG` class extends the 2D support to
+include the concept of a coordinate system, allowing the independent
+axis to be one of:
+
+- ``logical``
+- ``image``
+- ``world``
+
+where the aim is that the logical system refers to a pixel number (no
+coordinate system), image is a linear transform of the logical system,
+and world identifies a projection from the image system onto the
+celestial sphere. However, there is no requirement that this
+categorization holds as it depends on whether the optional
+:py:attr:`~sherpa.astro.data.DataIMG.sky` and
+:py:attr:`~sherpa.astro.data.DataIMG.eqpos` attributes are set when
+the :py:class:`~sherpa.astro.data.DataIMG` object is created.
+
+Using a coordinate system directly
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the independent axes are known, and not calculated via a coordinate
+transform, then they can just be set when creating the
+:py:class:`~sherpa.astro.data.DataIMG` object, leaving the
+:py:attr:`~sherpa.astro.data.DataIMG.coord` attribute set to
+``logical``.
+
+  >>> x0 = np.asarray([1000, 1200, 2000])
+  >>> x1 = np.asarray([-500, 500, -500])
+  >>> y = np.asarray([10, 200, 30])
+  >>> d = DataIMG("example", x0, x1, y)
```

Comment:
Usually, images are squares. Here, it's some kind of odd triangle. It's kind of cute that that works, but (a) I'm not sure that's a good idea in general and (b) I'm almost sure it's not a good idea to use that as the canonical example. Instead, I would move this example to the end of the discussion after the reader has seen the "normal" case or alternatively, add a sentence like "This is not the normal case for an image, hang on and see below"

Reply:
There is an interesting discussion to be had on what we mean for the `Data2D` and `DataIMG` cases - perhaps the `DataIMG` case should require a rectangular grid, but that's a different discussion than this PR.

I'll have to look at improving the docs tomorrow.

Reply:
Agreed. This is absolutely not what I had expected and I wonder if that's an accident of the design or if there is a real use case. Either way, this PR is certainly not the place to change that!

Reply:
I looked at replacing this with a grid, but I think it is important to let you know you don't need a grid. So I've added a note to the start of this little section pointing this out.